xrmocap/ops/bottom_up_association/matching/fourdag_matching.py

```diff
@@ -0,0 +1,770 @@
+                                if dist < self.max_epi_dist:
+                                    epi[joint1_candidate, joint2_candidate] = 1 - dist / self.max_epi_dist
+
+                        if self.normalize_edges:
+                            row_factor = np.clip(epi.sum(1), 1.0, None)
+                            col_factor = np.clip(epi.sum(0), 1.0, None)
+                            for i in range(len(row_factor)):
+                                epi[i] /= row_factor[i]
+                            for j in range(len(col_factor)):
+                                epi[:,j] /= col_factor[j]
+                        self.m_epiEdges[joint_id][view1][view2] = epi
+                        self.m_epiEdges[joint_id][view2][view1] = epi.T
+    
+    def cal_temp_edges(self):
+        for joint_id in range(self.n_joints):
+            for view in range(self.n_views):
+                rays = self.m_jointRays[view][joint_id]
+                if len(self.last_multi_kps3d) > 0 and len(rays) > 0:
+                    temp = np.full((len(self.last_multi_kps3d), len(rays)), -1.0)
+                    for pid, person_id in enumerate(self.last_multi_kps3d):
+                        skel = self.last_multi_kps3d[person_id]
+                        if skel[3, joint_id] > 0:
+                            for joint_candidate in range(len(rays)):
+                                dist = point2linedist(skel[:,joint_id][:3], self.cameras[view].Pos, rays[joint_candidate])
+                                if dist < self.max_temp_dist:
+                                    temp[pid, joint_candidate] = 1 - dist / self.max_temp_dist
+
+                    if self.normalize_edges:
+                        row_factor = np.clip(temp.sum(1), 1.0, None)
+                        col_factor = np.clip(temp.sum(0), 1.0, None)
+                        for i in range(len(row_factor)):
+                            temp[i] /= row_factor[i]
+                        for j in range(len(col_factor)):
+                            temp[:,j] /= col_factor[j]
+                    self.m_tempEdges[joint_id][view] = temp                    
+
+    def cal_bone_nodes(self):
+        for paf_id in range(self.n_pafs):
+            joint1, joint2 = self.paf_dict[0][paf_id], self.paf_dict[1][paf_id]
+            for view in range(self.n_views):
+                self.m_boneNodes[paf_id][view] = []
+                for joint1_candidate in range(len(self.kps2d_paf[view]['joints'][joint1])):
+                    for joint2_candidate in range(len(self.kps2d_paf[view]['joints'][joint2])):
+                        if self.kps2d_paf[view]['pafs'][paf_id][joint1_candidate, joint2_candidate] > 0:
+                            self.m_boneNodes[paf_id][view].append((joint1_candidate, joint2_candidate)) 
+
+    def cal_bone_epi_edges(self):
+        for paf_id in range(self.n_pafs):
+            joint_pair = [self.paf_dict[0][paf_id], self.paf_dict[1][paf_id]]
+            for view1 in range(5-1):
```

Comment:
We should use a variable representing the number of views, instead of `5`.

Reply:
Thanks, I will fix it